### DISCUSSION TITLE
Pulling metrics for kafka using Prometheus

### DISCUSSION THREAD

**Discussion by chary1112004** [strimzi/strimzi-kafka-operator#8485] [Q&A]
Hi,

We are trying to pull metric using prometheus. There is example for pod monitor in https://github.com/strimzi/strimzi-kafka-operator/blob/main/examples/metrics/prometheus-install/strimzi-pod-monitor.yaml

However after checking through pods, we do not see port **tcp-prometheus**. Does it requires to enable or added configuration in kafka operator and kafka cluster to expose metrics?

Any suggestion to resolve this issue would be very much appreciated.

Thanks!

**Comment by scholzj** [comment]
You need to enable the Prometheus metrics in the Strimzi custom resources (e.g. in the `Kafka` one for Kafka and ZooKeeper etc.) using the `metricsConfig` options: https://strimzi.io/docs/operators/latest/full/configuring.html#con-common-configuration-prometheus-reference ... you can also check the examples on GitHUb: https://github.com/strimzi/strimzi-kafka-operator/tree/main/examples/metrics

**Reply by chary1112004** [reply]
Hi @scholzj,

Thank you for the response. Follow this configuration https://github.com/strimzi/strimzi-kafka-operator/blob/main/examples/metrics/kafka-metrics.yaml then I could see data in grafana dashboard.

Thanks!
